### PR TITLE
Adding Tags Support to New-AzAppServicePlan,Set-AzAppServicePlan#9895 ; adding support to Stop executing if an exception is thrown when adding a custom domain to a website.#10422;Adding support to operations for App Services not in the same resource group as the App Service Plan #10482

### LINKED ISSUE #10422: webapp: Stop executing if an exception is thrown when adding a custom domain to a website.
## Description
When a user makes a mistake in their powershell script. It sometimes results in all their existing custom domains being removed. As part of the WebsiteClient in chaning customDomains the client does a GET to see existing domains, adds new domains, and then Deletes domains that are not included. If a customDomain fails to be added, we should not remove custom domains and should stop that part of the update customDomains code.

## Steps to reproduce
I understand that this isnt the newest version, but I forgot to save those commands
```powershell
$W=Get-AzureRmWebApp -ResourceGroupName "prod-rg" -Name "prod-rg"
$X = $W.hostnames
$X.add("mycustom.domain.com")
$HN = $X.tostring()
Set-AzureRmWebApp -ResourceGroupName "prod-rg" -Name "prod-rg" -HostNames  $HN

```
At this point, X = [oldDomain1, OldDomain2, New domain]
And HN = system.collections.generic.list`1[system.string]
When customer tries to send HN the websites cmdlet sends that one new domain, this fails because this 'domain' is not valid. and then the websites cmdlet continues 
## Environment data

```

```

## Module versions

```powershell

```

## Debug output

```

```

## Error output

```

```

### PR DESCRIPTION
**Pull request by ThejaChoudary**
Fixes #10422 , #10972 
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs [hunk 1]
```diff
@@ -63,6 +64,9 @@ public class NewAzureAppServicePlanCmdlet : AppServicePlanBaseCmdlet
         [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
         public SwitchParameter AsJob { get; set; }
 
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 7, Mandatory = false, HelpMessage = "Tags are name/value pairs that enable you to categorize resources")]
+        public IDictionary<string, string> Tags { get; set; }
```
Comment (markcowl) [inline src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs]: Parameter type shpould be Hashtable

File: src/Websites/Websites/Cmdlets/AppServicePlans/SetAzureAppServicePlan.cs [hunk 2]
```diff
@@ -49,6 +50,8 @@ public class SetAzureAppServicePlanCmdlet : AppServicePlanBaseCmdlet
         [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
         public SwitchParameter AsJob { get; set; }
 
+        [Parameter(ParameterSetName = ParameterSet1Name, Position = 7, Mandatory = false, HelpMessage = "Tags are name/value pairs that enable you to categorize resources")]
+        public IDictionary<string, string> Tags { get; set; }
```
Comment (markcowl) [inline src/Websites/Websites/Cmdlets/AppServicePlans/SetAzureAppServicePlan.cs]: Hashtable

File: src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs (hunk 1 above)
Comment (ThejaChoudary) [reply src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs]: As the Tags parameter type is Dictionary in Site Config,we declared it as the same.

File: src/Websites/Websites/Cmdlets/AppServicePlans/SetAzureAppServicePlan.cs (hunk 2 above)
Comment (ThejaChoudary) [reply src/Websites/Websites/Cmdlets/AppServicePlans/SetAzureAppServicePlan.cs]: As the Tags parameter type is Dictionary in Site Config,we declared it as the same.

Comment (markcowl) [reply src/Websites/Websites/Cmdlets/AppServicePlans/SetAzureAppServicePlan.cs]: The standard for this is to use Hashtable and convert to the wire type (usually ```Dictionary<string, string>```).  Please adhere to the standard

File: src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs (hunk 1 above)
Comment (markcowl) [reply src/Websites/Websites/Cmdlets/AppServicePlans/NewAzureAppServicePlan.cs]: The standard for this is to use Hashtable and convert to the wire type (usually ```Dictionary<string, string>```).  Please adhere to the standard

File: src/Websites/Websites.Test/Websites.Test.csproj [hunk 3]
```diff
@@ -23,8 +23,4 @@
     </None>
   </ItemGroup>
 
-  <ItemGroup>
-    <Folder Include="SessionRecords\Microsoft.Azure.Commands.Websites.Test.ScenarioTests.AccessRestrictionTests\" />
```
Comment (panchagnula) [inline src/Websites/Websites.Test/Websites.Test.csproj]: Don't remove this

File: src/Websites/Websites/Utilities/WebsitesClient.cs [hunk 4]
```diff
@@ -150,6 +150,7 @@ public void AddCustomHostNames(string resourceGroupName, string location, string
                 catch (Exception e)
                 {
                     WriteWarning("Could not set custom hostname '{0}'. Details: {1}", hostName, e.ToString());
+                    return;
```
Comment (panchagnula) [inline src/Websites/Websites/Utilities/WebsitesClient.cs]: Don't see a test case for this.

File: src/Websites/Websites.Test/ScenarioTests/WebAppTests.ps1 [hunk 5]
```diff
@@ -1159,6 +1160,17 @@ function Test-SetWebApp
...
+		$oldWebApp= Get-AzWebApp -ResourceGroupName $rgname -Name $webAppName
+		$CurrentWebApp = Set-AzWebApp -ResourceGroupName $rgname -Name $webAppName -HostNames $HN
+		
+		#Assert
+		$status
+		foreach($oldHN in $oldWebApp.HostNames)
+		{
+			Assert-True { $CurrentWebApp.HostNames -contains $oldHN }
+		}	
+
```
Comment (ThejaChoudary) [inline src/Websites/Websites.Test/ScenarioTests/WebAppTests.ps1]: Please find the above test case to handle script failure during adding the custom host names.

**Comment by ThejaChoudary** [comment]
Closing this pull request to raise a new one to avoid confusions

**Pull request closed without merging**